erts/emulator/beam/erl_hl_timer.c

```diff
@@ -2714,6 +2760,70 @@ erts_cancel_proc_timer(Process *c_p)
 			   (ErtsTimer *) tval);
 }
 
+void
+erts_pause_proc_timer(Process *c_p)
+{
+    erts_aint32_t prev_state = erts_atomic32_read_bor_acqb(&c_p->state,
+                                                           ERTS_PSFLG_PAUSED_TIMER);
+
+    ERTS_HLT_ASSERT(prev_state & ERTS_PSFLG_SUSPENDED);
+    if (!(prev_state & ERTS_PSFLG_PAUSED_TIMER)) {
+        ErtsPausedProcTimer *pptmr = create_paused_proc_timer(c_p);
+
+        if (!pptmr) {
+            return;
+        }
+
+        CANCEL_TIMER(c_p);
+
+        if (ERTS_PTMR_NONE != erts_atomic_cmpxchg_mb(&c_p->common.timer,
```

Comment:
Would a release barrier be sufficient here?

Reply:
I agree, changed to relb!